src/main/java/org/ibp/api/java/impl/middleware/germplasm/GermplasmExcelTemplateExportServiceImpl.java

```diff
@@ -189,13 +188,34 @@ private void writeObservationHeader(final HSSFSheet observationSheet, final bool
 			Map.Entry<String, ExcelCellStyle> entry = iterator.next();
 			HSSFCell cell = row.createCell(index, CellType.STRING);
 			cell.setCellStyle(this.sheetStylesMap.get(entry.getValue()));
-			cell.setCellValue(this.getMessageSource().getMessage(entry.getKey(), null, locale));
-			observationSheet.setColumnWidth(index, 13 * 250);
+			final String headerColumn = this.getMessageSource().getMessage(entry.getKey(), null, locale);
+			cell.setCellValue(headerColumn);
+			switch (entry.getKey()) {
+				case "export.germplasm.list.template.preferred.name.column":
+				case "export.germplasm.list.template.location.abbr.column":
+					observationSheet.setColumnWidth(index, 20 * 250);
+					break;
+				case "export.germplasm.list.template.entry.code.column":
+					observationSheet.setColumnWidth(index, 16 * 250);
```

Comment:
Looks ok. An improvement would be to rewrite IMPORT_HEADERS to be an enum, so that we don't repeat the string keys both here and and in the map static definition section, and we can switch over the enum